### ISSUE TITLE
make error : buildin LLVM step (projects compiler-rt)

### ISSUE THREAD

**Issue by bizulk** [oclint/oclint#301]
Hi !

Getting error when generating oclint at llvm installation step.

- Tried with last git mast, and 0.10.2 tag.
- Working in Debian Linux 8
- just call ./make in the script directory.

Here is the log :
`[ 20%] Built target clang_rt.lsan-x86_64`
`Scanning dependencies of target builtins`
`[ 20%] Built target builtins`
`Scanning dependencies of target profile`
`[ 20%] Built target profile`
`Scanning dependencies of target clang_rt.dfsan-x86_64-symbols`
`[ 20%] Generating exported symbols for clang_rt.dfsan-x86_64`
`Traceback (most recent call last):`
`  File "/home/tje/Work/OCLint/oclint-0.10.2/llvm/projects/compiler-rt/lib/sanitizer_common/scripts/gen_dynamic_list.py", line 110, in <module>
    main(sys.argv)`
 ` File "/home/tje/Work/OCLint/oclint-0.10.2/llvm/projects/compiler-rt/lib/sanitizer_common/scripts/gen_dynamic_list.py", line 103, in main`
`    print('  ' + f.encode('utf-8') + ';')`
`TypeError: Can't convert 'bytes' object to str implicitly`
`projects/compiler-rt/lib/dfsan/CMakeFiles/clang_rt.dfsan-x86_64-symbols.dir/build.make:53: recipe for target 'projects/compiler-rt/lib/dfsan/clang_rt.dfsan-x86_64.syms-stamp' failed`
`make[2]: *** [projects/compiler-rt/lib/dfsan/clang_rt.dfsan-x86_64.syms-stamp] Error 1`
`CMakeFiles/Makefile2:13087: recipe for target 'projects/compiler-rt/lib/dfsan/CMakeFiles/clang_rt.dfsan-x86_64-symbols.dir/all' failed`
`make[1]: *** [projects/compiler-rt/lib/dfsan/CMakeFiles/clang_rt.dfsan-x86_64-symbols.dir/all] Error 2`
`make[1]: *** Attente des tâches non terminées....`
`[ 20%] Building CXX object utils/TableGen/CMakeFiles/llvm-tblgen.dir/CodeGenMapTable.cpp.o`
`[ 20%] Building CXX object utils/TableGen/CMakeFiles/llvm-tblgen.dir/CodeGenRegisters.c`

How can I use older working version of LLVM, I don't understand the -llvm-branch option ?

**Comment by bizulk** [comment]
A LLVM bug was created : [24985](https://llvm.org/bugs/show_bug.cgi?id=24985)
But it is marked fixed.
I work in a VM and at first build all was find.
I switch from a 32bits to 64bits hosts and now I'm getting this errors.

**Comment by bizulk** [comment]
I finally found out that LLVM python scripts does not work with Python 3 & Python 3.4.
One should install Python 2.7 and make it as default.

I tried to used unsuccessfully an alias, so I finally set python2.7 default for system with the tool 'update-alternatives, as described [there](https://linuxconfig.org/how-to-change-from-default-to-alternative-python-version-on-debian-linux)

As this is not a oclint problem I close this issue.
